src/tools/clippy/clippy_lints/src/utils/mod.rs

```diff
@@ -323,6 +323,11 @@ pub fn implements_trait<'a, 'tcx>(
     trait_id: DefId,
     ty_params: &[GenericArg<'tcx>],
 ) -> bool {
+    // Do not check on infer_types to avoid panic in evaluate_obligation.
+    if ty.has_infer_types() {
+        return false;
+    }
+    let ty = cx.tcx.erase_regions(&ty);
```

Comment:
Isn't `src/tools/clippy` a separate repo?

Reply:
IIUC clippy is now subtree and we can make a change here not to break it, like #72466. cc @oli-obk right?

Reply:
correct